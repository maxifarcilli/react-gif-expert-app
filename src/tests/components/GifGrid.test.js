import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import GifGrid from '../../components/GifGrid';
import { useFetchGifs } from '../../hooks/useFetchGifs';

jest.mock('../../hooks/useFetchGifs');

describe('Pruebas sobre el componente <GifGrid />', () => {
   
    const categoria = 'test_category';

/*
    beforeEach( () => {
        jest.clearAllMocks(); // limpia cada simulacion en cada test.

        wrapper = 
    });
*/
    test('Debe de mostrar bien el componente <GifGrid />', () => {
        
        useFetchGifs.mockReturnValue({
            data: [],
            loading: true
        });

        const wrapper = shallow( 
            <GifGrid 
                key={categoria}
                category={categoria}
            /> 
        ); 

        expect( wrapper ).toMatchSnapshot();

    });
    
    test('Debe de mostrar items cuando se cargan imagenes con useFetchGifs', () => {
        
        const gifs = [{
            id: '123',
            url: 'https://localhost/hola.jpg',
            title: 'cualquier cosa'
        }];

        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false
        });

        const wrapper = shallow( 
            <GifGrid 
                key={categoria}
                category={categoria}
            /> 
        ); 
        
        expect( wrapper.find('p').exists() ).toBe(false);
        expect( wrapper.find('GifGridItem').length ).toBe( gifs.length );
        
        
    }); 

});
