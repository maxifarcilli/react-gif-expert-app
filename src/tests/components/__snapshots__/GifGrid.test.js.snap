// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas sobre el componente <GifGrid /> Debe de mostrar bien el componente <GifGrid /> 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    test_category
  </h3>
  <p>
    Cargando
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
